src/schema/njord.ts

```diff
@@ -70,6 +97,87 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         note,
       });
 
+      return { _: true };
+    },
+    awardPost: async (
+      _,
+      { productId, postId, note }: PostAwardInput,
+      ctx: AuthContext,
+    ) => {
+      // TODO add @auth directive
+      ctx.req.userId = '5GHEUpildSXvvbOdcfing';
+
+      const [product, post, userPost] = await queryReadReplica<
+        [
+          Pick<Product, 'type'>,
+          Pick<Post, 'id' | 'authorId'>,
+          Pick<UserPost, 'awardTransactionId'> | null,
+        ]
+      >(ctx.con, async ({ queryRunner }) => {
+        return Promise.all([
+          queryRunner.manager.getRepository(Product).findOneOrFail({
+            select: ['type'],
+            where: {
+              id: productId,
+            },
+          }),
+          queryRunner.manager.getRepository(Post).findOneOrFail({
+            select: ['id', 'authorId'],
+            where: {
+              id: postId,
+            },
+          }),
+          queryRunner.manager.getRepository(UserPost).findOne({
+            select: ['awardTransactionId'],
+            where: {
+              postId,
+              userId: ctx.userId,
+            },
+          }),
+        ]);
+      });
+
+      if (product.type !== ProductType.Award) {
+        throw new ForbiddenError('Can not gift this product');
+      }
+
+      if (!post.authorId) {
+        throw new ConflictError('Post does not have an author');
+      }
+
+      if (userPost?.awardTransactionId) {
+        throw new ConflictError('Post already awarded');
+      }
+
+      const transaction = await transferCores({
+        ctx,
+        receiverId: post.authorId,
+        productId,
+        note,
+      });
+
+      if (!transaction.productId) {
+        throw new Error('Product missing from transaction');
+      }
+
+      await ctx.con
+        .getRepository(UserPost)
+        .createQueryBuilder()
+        .insert()
+        .into(UserPost)
+        .values({
+          postId: post.id,
+          userId: ctx.userId,
+          awardTransactionId: transaction.id,
+          flags: {
+            awardId: transaction.productId,
+          },
+        })
+        .onConflict(
+          `("postId", "userId") DO UPDATE SET "awardTransactionId" = EXCLUDED."awardTransactionId", "flags" = user_post.flags || EXCLUDED."flags"`,
+        )
```

Comment:
Since UserPost can exist due to upvotes/downvotes on conflict we upsert the award columns